dspace-server-webapp/src/test/java/org/dspace/app/rest/PrimaryBitstreamControllerIT.java

```diff
@@ -0,0 +1,564 @@
+    @Test
+    public void testDeletePrimaryBitstreamCollectionAdmin() throws Exception {
+        // create new structure with Admin permissions on Collection
+        context.turnOffAuthorisationSystem();
+        Collection col2 = CollectionBuilder.createCollection(context, community).withAdminGroup(eperson).build();
+        Item item2 = ItemBuilder.createItem(context, col2).build();
+        Bundle bundle2 = BundleBuilder.createBundle(context, item2).withName("ORIGINAL").build();
+        Bitstream bitstream2 = createBitstream(bundle2);
+        bundle2.setPrimaryBitstreamID(bitstream2);
+        context.restoreAuthSystemState();
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(delete(getBundleUrl(bundle2.getID())))
+                        .andExpect(status().isNoContent());
+        // verify primaryBitstream was actually deleted
+        bundle2 = context.reloadEntity(bundle2);
+        Assert.assertNull(bundle2.getPrimaryBitstream());
+        // verify bitstream itself still exists
+        Assert.assertEquals(1, bundle2.getBitstreams().size());
+        Assert.assertEquals(bitstream2, bundle2.getBitstreams().get(0));
+    }
+
+    @Test
+    public void testDeletePrimaryBitstreamItemAdmin() throws Exception {
+        // create new structure with Admin permissions on Item
+        context.turnOffAuthorisationSystem();
+        Item item2 = ItemBuilder.createItem(context, collection).withAdminUser(eperson).build();
+        Bundle bundle2 = BundleBuilder.createBundle(context, item2).withName("ORIGINAL").build();
+        Bitstream bitstream2 = createBitstream(bundle2);
+        bundle2.setPrimaryBitstreamID(bitstream2);
+        context.restoreAuthSystemState();
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(delete(getBundleUrl(bundle2.getID())))
+                        .andExpect(status().isNoContent());
+        // verify primaryBitstream was actually deleted
+        bundle2 = context.reloadEntity(bundle2);
+        Assert.assertNull(bundle2.getPrimaryBitstream());
+        // verify bitstream itself still exists
+        Assert.assertEquals(1, bundle2.getBitstreams().size());
+        Assert.assertEquals(bitstream2, bundle2.getBitstreams().get(0));
+    }
+
+    @Test
+    public void testDeletePrimaryBitstreamForbidden() throws Exception {
+        bundle.setPrimaryBitstreamID(bitstream);
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(delete(getBundleUrl(bundle.getID())))
+                        .andExpect(status().isForbidden());
+    }
+
+    @Test
+    public void testDeletePrimaryBitstreamUnauthenticated() throws Exception {
+        bundle.setPrimaryBitstreamID(bitstream);
+
+        getClient().perform(delete(getBundleUrl(bundle.getID())))
+                        .andExpect(status().isUnauthorized());
+    }
+
+    private String getBundleUrl(UUID uuid) {
```

Comment:
it would be nice if you can rename the method in something like `getBundlePrimaryBitstreamLinkUrl`